src/todos/schemas/todos.schema.ts

```diff
@@ -21,7 +22,18 @@ export class Todo {
   updatedAt: Date
 }
 
-export const TodoSchema = SchemaFactory.createForClass(Todo)
+const TodoSchema = SchemaFactory.createForClass(Todo)
 
-// ? Question: is it an acceptable way to attach indexes here?
-TodoSchema.index({ userId: 1 })
+TodoSchema.post<TodoDocument>('save', async function (doc: TodoDocument) {
+  const UserModel: Model<UserDocument> = this.model('User')
+  const user = await UserModel.findById(doc.userId)
+
+  if (user) {
+    user.todos.unshift(doc)
```

Comment:
`.reduce` would be more effective.

Reply:
I tried to did it like this as one util for both "save" and "update" cases:
```
export const getLastUserTodos = (
  todos: Todo[],
  newTodo: Todo,
  todosCount = 5,
) => {
  const isNewTodoExist = todos.some((todo) => todo._id.equals(newTodo._id))
  const lastIndex = isNewTodoExist ? todosCount - 1 : todosCount - 2

  todos = todos.reduce(
    (todos, todo, index) => {
      if (index > lastIndex) return todos
      if (todo._id.equals(newTodo._id)) return todos
      todos.push(todo)
      return todos
    },
    [newTodo],
  )

  return todos
}
```
But I still think `reduce` here just adds complexity to a fairly simple task, and it can be reworked in a more compact way like this:
```
export const getLastUserTodos = (
  todos: Todo[],
  newTodo: Todo,
  todosCount = 5,
) => {
  const filteredTodos = todos.filter((todo) => !todo._id.equals(newTodo._id))
  return [newTodo, ...filteredTodos].slice(0, todosCount)
}
```
I tested both of them in possible usecases, both works fine. Of course, `reduce` gives us more space to add some extra logic in the future, but for now it looks overly complex for the current requirement.